lib/bibliothecary/analyser.rb

```diff
@@ -130,25 +130,42 @@ def analyse_file_info(file_info_list)
           .select(&method(:match_info?))
 
         matching_info.map do |info|
-          # TODO: Return multiple analysis for one info
-          analyse_contents_from_info(info)
-            .merge(related_paths: related_paths(info, matching_info))
+          analyse_contents_from_info(info).map do | analysis |
+            analysis.merge(related_paths: related_paths(info, matching_info))
+          end
         end
       end
 
       def analyse_contents(filename, contents)
         analyse_contents_from_info(FileInfo.new(nil, filename, contents))
       end
 
+
       def analyse_contents_from_info(info)
+        # If this Package Manager is a multiple, analyse multiple, otherwise just analyse one,
+        determine_multiple_from_info(info) ? analyse_multiple(info) : analyse_one(info)
+      end
+
+      def analyse_one(info)
         kind = determine_kind_from_info(info)
         dependencies = parse_file(info.relative_path, info.contents)
 
-        Bibliothecary::Analyser::create_analysis(platform_name, info.relative_path, kind, dependencies || [])
+        [create_analysis_for_dependencies(dependencies, info, kind)]
       rescue Bibliothecary::FileParsingError => e
         Bibliothecary::Analyser::create_error_analysis(platform_name, info.relative_path, kind, e.message)
       end
 
+      def analyse_multiple(info)
+        # This method allows multiple analyses to come out
+        # set the mapping `multiple: true` and an array of `kinds`
+        kinds = determine_kind_from_info(info)
+        dependencies = parse_file(info.relative_path, info.contents)
+
+        kinds.map { |kind| create_analysis_for_dependencies(dependencies[kind], info, kind) }
+      rescue Bibliothecary::FileParsingError => e
```

Comment:
I wonder if this should be inside the map (so for example we could get the conda manifest but throw an error on the lockfile resolution, in principle). For the way things work now I guess it couldn't happen, but it seems conceptually sound to have this rescue inside the map around each create_analysis_for_dependencies call ?

Reply:
This is why they pay you the big money! This makes so much more sense.